### PR TITLE
Validate nimi-python system tests on the dynamic VM's hosted on AWS using GitHub actions

### PR DESCRIPTION
**Pull request by SukruthGrandhiNI**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

- [ ] ~~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~

- [ ] ~~I've added tests applicable for this pull request~~

### What does this Pull Request accomplish?

In this Pull Request added a functionality to validate system tests on AWS VM's using GitHub action. 

Reason for using GitHub actions: External contributors can view logs for the pipeline runs using GitHub actions. 
Earlier MTO validated system tests using VM's hosted on NI network + Azure pipelines. Please refer PR #1653 for more details.

Feature specifications.

1. System tests are run on the dynamic created VM's on AWS using GitHub actions. This means pay as you use.
2.  There are 3 workflows
     a. nimibot_rhel_64: This workflow validates system tests on 64bit Python version on RHEL. There are only two modules(nidmm and niswithc) are validated as part of testing.
     b. nimibot_win_32: This workflow validates system tests on 32bit Python version on Windows 10.
     c. nimibot_win_64: This workflow validates system tests on 64bit Python version on Windows 10.
         9 modules are validates as part of windows system runs.
3.   Each module system tests are run on individually dynamically created VM's. That means  2(rhel_64) + 9(win_32) + 9(win_64, total 20 VM's are dynamically created for PR testing.

Next improvements:
1. Cost analysis of adding support for multiple agents on a single VM. Since multiple agents need to run on single VM, VM configuration needs update. Analyze the cost of using multiple low configuration VM's(single agent) V/S high configuration single VM(multiple agents).

Next set of activities.
1. Update the wiki documentations.
2. Administrator should control the GitHub workflows for the external contributors. i.e  External contributor PR validation will happen only after the approval of nimi-python Admin.

Additional technical details.
Created 1 webhook for each workflow. Total 3 webhook's are created for 3 workflow.
image

### List issues fixed by this Pull Request below, if any.

Allow Pipeline logs for external contributors.

### What testing has been done?

nimibot_rhel_64 pipeline workflow: [link](https://github.com/ni/nimi-python/actions/runs/2088757199)
nimibot_win_32  pipeline workflow: [link](https://github.com/ni/nimi-python/actions/runs/2088757201)
nimibot_win_64 pipeline workflow: [link](https://github.com/ni/nimi-python/actions/runs/2088757200)

### CODE REVIEW

File: .github/actions/linux/action.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  using: "composite"
+  steps:
+    - run: |
+        source /etc/bashrc
+        cd generated
+        mkdir junit
+        mkdir kibana
+        cd ${{ inputs.module_name }}
+        python -m tox -c tox-system_tests.ini
+      shell: bash
```
Comment (marcoskirsch) [inline .github/actions/linux/action.yml:19]: this file is identical to the windows one except for the shell.

Could you have one file and make that an input like module_name?
Could you then move the file up one directory and rename to run_system_tests_action.yml or something like that?

**Review by marcoskirsch — CHANGES_REQUESTED** [review]

**Comment by marcoskirsch** [comment]
It's not a lot of code in total, but I wonder how we could reduce duplication.

File: .github/actions/linux/action.yml (hunk 1 above)
Comment (SukruthGrandhiNI) [reply .github/actions/linux/action.yml:19]: Could you have one file and make that an input like module_name?
It's a good point but I could not combine in single file, github action composites does not have flexibity unlike azure pipelines.

Could you then move the file up one directory and rename to run_system_tests_action.yml or something like that?
github composite action only understands action.yml so we cannot rename to run_system_tests_action.

File: .github/actions/linux/action.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  module_name:
+    description: "module name"
+    required: true
+    default: ""
+
+runs:
+  using: "composite"
+  steps:
+    - run: |
+        source /etc/bashrc
```
Comment (SukruthGrandhiNI) [inline .github/actions/linux/action.yml:13]: @marcoskirsch  this line is needed only for Linux.

**Review by marcoskirsch — APPROVED** [review]

**Review by sbethur — APPROVED** [review]

**Pull request merged**